docs/platform/porting/system.md

```diff
@@ -2157,3 +2157,382 @@ $ sensorctl test accelerometer 100 1000 /* enable accelerometer with interval 10
+    - System disk space usage
+    - System memory statistics
+-   Process information:
+    - Process information by `ps` command
+-   Device information:
+    - Device major numbers from `/proc/devices` 
+
+
+The dump contents of **dump_systemstate** can be modified by the following command line options:
+
+Option | Description
+-------|----------------------------------
+-f     | write to file (instead of stdout)
+-k     | dump kernel messages (root only)
+-d     | dump dlog messages
+-j     | dump journal log messages
+-p     | dump list of installed packages
+-e     | dump extras defined in the config
+
+Depending on the command line options, **dump_systemstate** can save the output of these commands:
+
+-   `/bin/df -h`
+-   `/bin/du -ah /opt --exclude=/opt/usr`
+-   `/bin/ls -al /opt/etc/localtime`
+-   `/bin/top -bcH -n 1`
+-   `/bin/ps auxfw`
+-   `/bin/memps -v`
+-   `/bin/buxton2ctl dump memory`
+-   `/bin/buxton2ctl dump system`
+-   `/usr/bin/pkgcmd -l`
+-   `/bin/dmesg -T`
+-   `/bin/dlogutil -d -v threadtime -u 16384`
+-   `/bin/journalctl -b -n 1024`
+
+
+There are two types of configuration files, `files/` subdirectory and `programs/` subdirectory. These files allow to specify the results of additional files and program that will be included in the **dump_systemstate** output:
+
+-   `files/` subdirectory: Specified files will be appended to the **dump_systemstate** output. File contains:
+    - **[SECTION_NAME]**
+    - **title**: Title will be printed before contents of file
+    - **path**: Path to the file
+
+    Example:
+
+    ```
+      [UNIQUE_ID_KEY]
+      title=header line that gets printed (path gets appended too)
+      path=/path/to/the/file
+
+      [DLOG_CONF]
+      title=dlog configuration file
+      path=/opt/etc/dlog.conf
+    ```
+
+-   `programs/` subdirectory: Specified commands will be executed and the output will be added to the **dump_systemstate** output. File contains:
+    - **[SECTION_NAME]**
+    - **title**: Title that will be printed before contents of the file
+    - **path**: Path to the application
+    - **args**: Arguments for the application
+    - **env**: Environment variable which should be passed to the application
+
+    Example:
+
+    ```
+      [UNIQUE_ID_KEY]
+      title=header line describing the program (will be printed alongside env, path and args)
+      path=/path/to/the/program/executable
+      args=-x foo --verbose
+      env=POSIXLY_CORRECT=1
+
+      [DLOG_DUMP]
+      title=dump dlog contents
+      path=/usr/bin/dlogutil
+      args=-d
+    ```
+
+### livedumper
+
+**livedumper** is a tool to save the core dump of the live process. Livedumper sends `SIGSTOP` to the process before dumping and `SIGCONT` after dumping.
+
+Usage:
+
+Option         | Description
+---------------|-------------------------------------------------------------------------------
+-f <file_name> | Saves core dump to the specified file (default livecore.<PID> in current directory)
```

Comment:
@mijin85cho are you sure you want a HTML specific characters in this document?

Reply:
@mmoscicki2 Basically we request to follow the markdown syntax. 
However in case of some strings or tables that can't be expressed by markdown, we take HTML's.

As I commented, for now \<PID> and \<fd> can't be expressed as they are considered as some kind of tag. To prevent it, I requested to use \&lt; and \&gt;